### PR TITLE
Support for TCP type & ExposedPort on `azurerm_container_app`

### LINKED ISSUE #23738: Support for ExposedPort on azurerm_container_app
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Description

Currently the exposed port on azurerm_container_app is missing. It is needed when you want to expose a different port than the target port. It is available if the transport is TCP. It is already supported but not documented (would be nice to document it). It was made possible when the API version was updated in #22804.

image

The latest version of the SDK is being used and the field is available already https://github.com/hashicorp/go-azure-sdk/blob/2f16ee0d84fbe0ebc235a86357d6b421190f6736/resource-manager/containerapps/2023-05-01/containerapps/model_ingress.go

### New or Affected Resource(s)/Data Source(s)

azurerm_container_app

### Potential Terraform Configuration

```hcl
resource "azurerm_container_app" "example" {
  name                           = "${var.resource_prefix}-example"
  container_app_environment_id   = azurerm_container_app_environment.example.id
  resource_group_name            = azurerm_resource_group.example.name
  revision_mode                  = "Single"

  ingress {
    external_enabled = true
    target_port      = 9000
    exposed_port     = 9090
    transport        = "tcp"
  }
}
```

### References

_No response_

### PR DESCRIPTION
**Pull request by SpartakusMd**
- Added the missing doc that TCP type is supported. Result of #22804
- Added missing option for ExposedPort

Fixes #23738 #21133 #22755

### CODE REVIEW

File: internal/services/containerapps/container_app_resource_test.go [hunk 1]
```diff
@@ -264,6 +264,21 @@ func TestAccContainerAppResource_completeUpdate(t *testing.T) {
 	})
 }
 
+func TestAccContainerAppResource_completeTcpExposedPort(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_container_app", "test")
+	r := ContainerAppResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.complete(data, "rev1"),
```
Comment (stephybun) [inline internal/services/containerapps/container_app_resource_test.go:273]: This should probably be using the test config added in this PR 
```suggestion
			Config: r.completeTcpExposedPort(data, "rev1"),
```

Comment (SpartakusMd) [reply internal/services/containerapps/container_app_resource_test.go]: Nice catch

Comment (stephybun) [reply internal/services/containerapps/container_app_resource_test.go]: Unfortunately we have a test failure @SpartakusMd 
```
------- Stdout: -------
=== RUN   TestAccContainerAppResource_completeTcpExposedPort
=== PAUSE TestAccContainerAppResource_completeTcpExposedPort
=== CONT  TestAccContainerAppResource_completeTcpExposedPort
    testcase.go:113: Step 1/2 error: Error running apply: exit status 1
        Error: creating Container App (Subscription: "*******"
        Resource Group Name: "acctestRG-CAEnv-231101113719484045"
        Container App Name: "acctest-capp-231101113719484045"): performing CreateOrUpdate: containerapps.ContainerAppsClient#CreateOrUpdate: Failure sending request: StatusCode=0 -- Original Error: Code="ContainerAppInvalidIngressAllowInsecureForTcpApp" Message="Allow insecure can only be set for http transport."
          with azurerm_container_app.test,
          on terraform_plugin_test.tf line 130, in resource "azurerm_container_app" "test":
         130: resource "azurerm_container_app" "test" {
        creating Container App (Subscription: "*******"
        Resource Group Name: "acctestRG-CAEnv-231101113719484045"
        Container App Name: "acctest-capp-231101113719484045"): performing
        CreateOrUpdate: containerapps.ContainerAppsClient#CreateOrUpdate: Failure
        sending request: StatusCode=0 -- Original Error:
        Code="ContainerAppInvalidIngressAllowInsecureForTcpApp" Message="Allow
        insecure can only be set for http transport."
--- FAIL: TestAccContainerAppResource_completeTcpExposedPort (514.69s)
FAIL
```

**Review by stephybun — COMMENTED** [review]
New test failure 🙃  - I'm no SME on ContainerApps so I'm not sure how specific the configuration will need to be in order to use `exposed_port` or how much validation we should be putting into the resource to prevent users from running into this kind of error. 

@jackofallops perhaps you could provide a little bit of guidance here?
```
------- Stdout: -------
=== RUN   TestAccContainerAppResource_completeTcpExposedPort
=== PAUSE TestAccContainerAppResource_completeTcpExposedPort
=== CONT  TestAccContainerAppResource_completeTcpExposedPort
    testcase.go:113: Step 1/2 error: Error running apply: exit status 1
        Error: creating Container App (Subscription: "*******"
        Resource Group Name: "acctestRG-CAEnv-231101121940949762"
        Container App Name: "acctest-capp-231101121940949762"): performing CreateOrUpdate: containerapps.ContainerAppsClient#CreateOrUpdate: Failure sending request: StatusCode=0 -- Original Error: Code="ContainerAppTcpRequiresVnet" Message="Applications with external TCP ingress can only be deployed to Container App Environments that have a custom VNET. Set ingress traffic to 'Limited to Container Apps Environment' if you want to deploy to a Container App Environment without a custom VNET. For more information on custom VNETs, please visit https://go.microsoft.com/fwlink/?linkid=2238174
          with azurerm_container_app.test,
          on terraform_plugin_test.tf line 130, in resource "azurerm_container_app" "test":
         130: resource "azurerm_container_app" "test" {
        creating Container App (Subscription: "*******"
        Resource Group Name: "acctestRG-CAEnv-231101121940949762"
        Container App Name: "acctest-capp-231101121940949762"): performing
        CreateOrUpdate: containerapps.ContainerAppsClient#CreateOrUpdate: Failure
        sending request: StatusCode=0 -- Original Error:
        Code="ContainerAppTcpRequiresVnet" Message="Applications with external TCP
        ingress can only be deployed to Container App Environments that have a custom
        VNET. Set ingress traffic to 'Limited to Container Apps Environment' if you
        want to deploy to a Container App Environment without a custom VNET. For more
        information on custom VNETs, please visit
        https://go.microsoft.com/fwlink/?linkid=2238174
--- FAIL: TestAccContainerAppResource_completeTcpExposedPort (1162.14s)
FAIL
```

**Comment by jackofallops** [comment]
> New test failure 🙃 - I'm no SME on ContainerApps so I'm not sure how specific the configuration will need to be in order to use `exposed_port` or how much validation we should be putting into the resource to prevent users from running into this kind of error.
> 
> @jackofallops perhaps you could provide a little bit of guidance here?


@SpartakusMd / @stephybun - The config is missing a user defined subnet to attach to the Container App Environment, using `templateWithVnet` in place of `templatePlusExtras` I think will resolve that. fwiw, a lot of the config in the `completeTcpExposedPort` test can be removed as it only needs to test the new property / ingress block (e.g. the extended `container` config for `probes` and `volumes`, since they're unrelated to the change and provide no additional benefit here)

**Comment by SpartakusMd** [comment]
Switched to use `templateWithVnet ` and did some cleanup. Hopefully it's enough 🤞

**Review by stephybun — COMMENTED** [review]
@SpartakusMd third times a charm - test passed! 🎉 

Just need to fix up the terraform linting that the GitHub Action is flagging then this should be good to go!

**Comment by SpartakusMd** [comment]
Perfect. Fixed the linting too.

**Review by stephybun — APPROVED** [review]
Thanks @SpartakusMd LGTM 💯

**Pull request merged**